### PR TITLE
chore: add oauth2 proxy provider backend module

### PR DESCRIPTION
**Pull request by djamaile**
#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by djamaile** [comment]
I'll probably have time after wellness weak & BackstageCon to pick this up further. Seems generating the api reports is error'ing on packages that weren't changed.

File: plugins/auth-backend-module-oauth2-proxy-provider/src/resolvers.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  createSignInResolverFactory,
+  SignInInfo,
+} from '@backstage/plugin-auth-node';
+import { OAuth2ProxyResult } from '@backstage/plugin-auth-backend';
+
+/**
+ * @public
+ */
+export namespace oauth2ProxySignInResolvers {
+  export const forwardedUserMarchingUserEntityName =
```
Comment (Rugvip) [inline plugins/auth-backend-module-oauth2-proxy-provider/src/resolvers.ts:27]:
```suggestion
  export const forwardedUserMatchingUserEntityName =
```

😁

**Review by Rugvip — COMMENTED** [review]
Nice!: tada

File: packages/backend/package.json [hunk 2]
```diff
@@ -35,6 +35,7 @@
     "@backstage/plugin-adr-backend": "workspace:^",
     "@backstage/plugin-app-backend": "workspace:^",
     "@backstage/plugin-auth-backend": "workspace:^",
+    "@backstage/plugin-auth-backend-module-oauth2-proxy-provider": "workspace:^",
```
Comment (vinzscam) [inline packages/backend/package.json:38]: this shouldn't be needed, right? 🤔 

```suggestion
```

File: plugins/auth-backend-module-oauth2-proxy-provider/package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    "test": "backstage-cli package test",
+    "clean": "backstage-cli package clean",
+    "prepack": "backstage-cli package prepack",
+    "postpack": "backstage-cli package postpack"
+  },
+  "dependencies": {
+    "@backstage/backend-common": "workspace:^",
+    "@backstage/backend-plugin-api": "workspace:^",
+    "@backstage/errors": "workspace:^",
+    "@backstage/plugin-auth-backend": "workspace:^",
```
Comment (vinzscam) [inline plugins/auth-backend-module-oauth2-proxy-provider/package.json:29]: I see there is a circular dependency between `auth-backend-module-oauth2-proxy-provider` and `plugin-auth-backend`, which right now depend on each other

**Review by vinzscam — CHANGES_REQUESTED** [review]
nice!

File: packages/backend/package.json (hunk 2 above)
Comment (djamaile) [reply packages/backend/package.json]: Fixed in https://github.com/backstage/backstage/pull/20685/commits/f88b2da6c9741d35234d31cdfef9e305a4f0b624

**Review by Rugvip — APPROVED** [review]
Nice, thank you! 😁 

Let's :shipit:

**Pull request merged**